### PR TITLE
docs: Add upgrading doc

### LINKED ISSUE #78: Add an upgrading doc
Add a doc explaining how to "upgrade" from Clear Containers. At this stage, that will actually just be "install kata packages" :smile:

### PR DESCRIPTION
**Pull request by jodh-intel**
Add a document explaining how to upgrade a system:

- Running Clear Containers.
- Already installed with Kata Containers binaries.

Fixes #78.

Signed-off-by: James O. D. Hunt <james.o.hunt@intel.com>

### CODE REVIEW

**Comment by jodh-intel** [comment]
@egernst, @jcvenegas, @klynnrif - ptal.

@bergwolf, @gnawux, @laijs - As the summary shows, this is (mostly) Clear Containers specific. I am happy to add `runv` details if you can supply me with details ;) Alternatively, do you want to add those details in a follow-up PR?

**Comment by bergwolf** [comment]
There is simply no such thing as upgrading from runv. kata is a fresh installation and it can even run together with runv without affecting each other.

**Comment by jodh-intel** [comment]
Hi @bergwolf - I thought that would be the case. Are you happy for this doc to land and then potentially for someone to raise a follow-on PR to update this doc explaining the runv approach?

**Comment by bergwolf** [comment]
@jodh-intel yes, that's fine with me. I'll raise a PR to add the runv approach after this is landed.

**Comment by jodh-intel** [comment]
Great - thanks @bergwolf.

**Comment by jcvenegas** [comment]
@jodh-intel  looks good - I'll be giving a try today to make sure it works in my clear-containers dev machine.

**Comment by egernst** [comment]
Assuming someone (@jcvenegas ) tests/verifies, the text and plan looks great to me.

File: Upgrading.md [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+Kata Containers requires additional resources to create a virtual machine
+container. These resources are called
+[Kata Containers assets](https://github.com/kata-containers/documentation/blob/master/architecture.md#assets)
+which comprise a guest kernel and a root filesystem or initrd image. This
+section describes when these components are updated.
+
+Since the official assets are packaged, they are automatically upgraded when
+new package versions are published.
+
+> **Warning**: Note that if you are using custom assets (by modifying the
```
Comment (klynnrif) [inline Upgrading.md]: **Warning**: Note that if you use custom assets (by modifying the

File: Upgrading.md [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+> [Kata Runtime configuration > file](https://github.com/kata-containers/runtime/#configuration)),
+> it is your responsibility to ensure they are updated as necessary.
+
+### Guest kernel
+
+The `kata-linux-container` package contains a Linux\* kernel based on the
+latest vanilla version of the
+[longterm kernel](https://www.kernel.org/)
+and including a small number of
```
Comment (klynnrif) [inline Upgrading.md]: Should this be "and include"?

File: Upgrading.md [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+[patches](https://github.com/kata-containers/packaging/tree/master/kernel).
+
+The `Longterm` branch is only updated with
+[important bug fixes](https://www.kernel.org/category/releases.html)
+meaning this package will only be updated when necessary.
```
Comment (klynnrif) [inline Upgrading.md]: meaning this package is only updated when necessary.

File: Upgrading.md [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+
+The guest kernel package will be updated when a new longerm kernel is released
```
Comment (klynnrif) [inline Upgrading.md]: Rewrite + missing "t" in "longterm": The guest kernel package is updated when a new longterm kernel is released

**Comment by jodh-intel** [comment]
Thanks @klynnrif - branch updated.

**Pull request merged**